probe-rs/examples/ram_download.rs

```diff
@@ -0,0 +1,165 @@
+use probe_rs::{
+    config::registry::{Registry, SelectionStrategy},
+    coresight::memory::MI,
+    probe::{daplink, stlink, DebugProbe, DebugProbeType, MasterProbe, WireProtocol},
+    session::Session,
+    target::info::ChipInfo,
+};
+
+use std::num::ParseIntError;
+use std::time::Instant;
+
+use pretty_env_logger;
+use rand::prelude::*;
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+struct CLI {
+    #[structopt(long = "target")]
+    target: Option<String>,
```

Comment:
We should change this to chip as this is what's used in the rest of the APIs/code.